### ISSUE TITLE
accessibilityActions with custom actions on Android seems to be broken (announcing "Action not supported" with example copied from docs)

### ISSUE THREAD

**Issue by BogdanCerovac** [facebook/react-native#47268]
### Description

Updated React Native version to 0.74.5 (default in Expo Snack with SDK 51.0.0)

By using TalkBack screen-reader I tired to check out how the accessibilityActions works on my device and it seems that they don't. At least not via Expo (using snacks for agility, I guess it should be the same as if I used it via local environment (?)).

I've tested this with two different physical devices and got same result (Both Android 14 with last official updates, Samsung Galaxy S22 (Accessibility Suit V.  15.1.00.7) and Motorola Moto G24 (Accessibility Suit V. 15.0.1668772353))...

Copied example from docs (https://reactnative.dev/docs/accessibility#accessibility-actions), to test it via Expo snack, with TalkBack screen reader and both custom actions failed to trigger their function (announcing "Action not supported").

Additionally it seems that it's just possible to reach two of them in the menu (I couldn't actually reach all three of them, only two of them were available via swiping up/down (perhaps there is a special gesture, but I would expect swiping "in circles" would allow me to reach all three).

Please check the example by yourself or the video that shows visually what I wrote here and in the steps to reproduce.

### Steps to reproduce

1. Open expo link with your Android device
2. Enable TalkBack screen reader
3. Swipe to the control
4. Try to swipe up and down when control is in focus
5. Try to activate the function by double tapping the screen

### React Native Version

0.74.5

### Affected Platforms

Runtime - Android

### Output of `npx react-native info`

```text
N.A.
```

### Stacktrace or Logs

```text
N.A.
```

### Reproducer

https://snack.expo.dev/yFTlDxsuMqyrNYqTSslN8

### Screenshots and Videos

https://youtu.be/-1nAjxijIrc

**Comment by BogdanCerovac** [comment]
Not clear to me if Expo snacks supports latest version of React Native, but I've checked the current version and it's 0.74.5 (therefore the edit of OP)

**Comment by BogdanCerovac** [comment]
@cortinico - Thanks for quick reply. 

I did now and it seems to be the same as with Expo snack;

Here is the demo: https://youtu.be/HKdpSXO7qW4

And here is the code part that I tested:

```
<View
        style={{borderWidth:1, padding:20,margin:50,borderRadius:10}}
        accessible={true}
        accessibilityLabel="Interactive"
        accessibilityActions={[
          {name: 'activate', label: 'activate'},
          {name: 'longpress', label: 'longpress'},
          {name: 'cut', label: 'cut'},
          {name: 'copy', label: 'copy'},
          {name: 'paste', label: 'paste'},
        ]}
        onAccessibilityAction={event => {
          switch (event.nativeEvent.actionName) {
            case 'activate':
              Alert.alert('Alert', 'activate action success');
              break;
            case 'longpress':
              Alert.alert('Alert', 'longpress action success');
              break;
            case 'cut':
              Alert.alert('Alert', 'cut action success');
              break;
            case 'copy':
              Alert.alert('Alert', 'copy action success');
              break;
            case 'paste':
              Alert.alert('Alert', 'paste action success');
              break;
          }
        }}
      >

        <Text>Interactive</Text>
      </View>
```
(Basically the same as in the Expo snack)

**Comment by TOPans** [comment]
@BogdanCerovac 
Hi, any ideas on how to fix this, I encountered the same problem on a project.
tried a more recent version of react native 75.4, similar problem.

**Comment by BogdanCerovac** [comment]
@TOPans - I guess our best bet would be to debug if RN is using the correct Android mappings, but I am not familiar with the implementation and I unfortunately don't have enough Android experience to do that myself, sorry.

If I am not mistaken the RN also planned to include some testing around accessibility actions (https://github.com/facebook/react-native/blob/fcf3c8cab7568f583c26aacb398f8e8a61da80eb/packages/react-native/types/__typetests__/index.tsx#L2028), but it seems to me that it's not called / used...

**Comment by mateoguzmana** [comment]
I'm unable to reproduce this in the latest version. Could you try reproducing this in 0.76? good to mention I just used the Android simulator to check this.

<details>
<summary>Screenshots</summary>
<p align="center">
  <img src="https://github.com/user-attachments/assets/15ede408-8e32-4494-b1e1-83c48bd38895" width="40%" />
  <img src="https://github.com/user-attachments/assets/5a6d72d8-a16c-42a8-bbdb-f6b06c4b63f7" width="40%" />
</p>
</details>

I noticed the `active` and `longpress` are not present as actions but from the official example only the 3 options shown in the screenshot are present so that might be expected. Also, the `active` option worked but it is just not present in the menu.

**Comment by BogdanCerovac** [comment]
@mateoguzmana  - thanks for letting me know.

I just tested it via Expo (v52.0.0.0) and can confirm it works there.

Test snack URL:  https://snack.expo.dev/opFQuARK-XzfQR7NCpvBf

Demo video: https://youtu.be/kSH2LnGCwHk

Code for reference if snack will be removed:

```
import { Text, SafeAreaView, StyleSheet, View, Pressable, Linking, Alert  } from 'react-native';


export default function App() {
  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.paragraph} accessibilityRole="header">
        Testing accessibilityActions
      </Text>
<Text>Testing accessibilityActions based on the official docs (<Pressable style={{display:"inline-block"}} accessibilityRole="link" accessibilityLabel="https://reactnative.dev/docs/accessibility#accessibility-actions" onPress={() => Linking.openURL('https://reactnative.dev/docs/accessibility#accessibility-actions')}>
      <Text style={{ color: 'blue', textDecorationLine: 'underline' }}>
        https://reactnative.dev/docs/accessibility#accessibility-actions
      </Text>
    </Pressable>)</Text>

      <View
        style={{borderWidth:1, padding:5,margin:5,borderRadius:10}}
        accessible={true}
        accessibilityLabel="Actions available!"
        accessibilityActions={[
          {name: 'activate', label: 'activate'},
          {name: 'longpress', label: 'longpress'},
          {name: 'cut', label: 'cut'},
          {name: 'copy', label: 'copy'},
          {name: 'paste', label: 'paste'},
          {name: 'additional', label: 'additional'},
        ]}
        onAccessibilityAction={event => {
          switch (event.nativeEvent.actionName) {
            case 'activate':
              Alert.alert('Alert', 'activate action success');
              break;
            case 'longpress':
              Alert.alert('Alert', 'longpress action success');
              break;
            case 'cut':
              Alert.alert('Alert', 'cut action success');
              break;
            case 'copy':
              Alert.alert('Alert', 'copy action success');
              break;
            case 'paste':
              Alert.alert('Alert', 'paste action success');
              break;
             case 'additional':
              Alert.alert('Alert', 'additional action success');
              break;
          }
        }}
      ><Text>Actions available!</Text></View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    backgroundColor: '#ecf0f1',
    padding: 8,
  },
  paragraph: {
    margin: 24,
    fontSize: 18,
    fontWeight: 'bold',
    textAlign: 'center',
  },
});

```

**Comment by Rotyslav-Liashenko** [comment]
Hello. I also encountered this problem. React Native version: 0.76.7. I am using react-native without expo

**Comment by Rotyslav-Liashenko** [comment]
@BogdanCerovac I opened your example through your snack. When I swipe up or down, I see the actions, but when I double tap with the action selected, it says the action is not supported. Do you have the same behavior?

**Comment by Rotyslav-Liashenko** [comment]
It's strange, when I do a three-finger tap and see the action menu, everything works. But when I switch actions by swiping up or down and double-tap, it says "Action not supported". How do I make the switch by swiping up or down work and the desired action is performed. Could you help me? @BogdanCerovac @mateoguzmana Sorry to bother you, but I can't find any information about this anywhere.

**Comment by BogdanCerovac** [comment]
@Rotyslav-Liashenko - it's indeed still a problem when just swiping, reopening this issue as I only tested it from the actions menu.

Tested with just swiping on an native-native app (Gmail to be precise) and there it works well there, so this is still a (partial) bug.

Sorry, don't have more information at hand (not an Android dev), I guess somebody with Android experience needs to dive into the native part to figure this out.

**Comment by Doush-H** [comment]
Hello, we are facing the same issue in our 10M+ user-based app. Currently, we are working on our accessibility because of the EAA requirements that we're trying to meet. We are not using Expo at the moment, and honestly, the phrase "it works on Expo" doesn't feel like a solution. Are there any updates on this topic yet?

**Comment by Laksuh** [comment]
We encounter the same issue. Are there any updates yet?

**Comment by a-klotz-p8** [comment]
This is still a issue in RN 0.78 and considering many devs need to implement the EAA until 28 June 2025. This seems like something that needs to be fixed asap.

**Comment by BogdanCerovac** [comment]
@Doush-H  - absolutely agree that "it works on Expo" is not a solution. It must work on React Native and it needs to stay working after they update versions. It seems that the solution is in the Android source code, unfortunately I am not an Android developer, so can't help here.

Do you have somebody in the team that can check the Android source/bridge to JS here and make a pull request? As it seems that this issue is ignored by the core team (among other accessibility issues also vital for European Accessibility Act)...

**Comment by oaron** [comment]
I’ve spent several days investigating this issue. I attempted to write a patch and even modified some of the React Native Java files, but unfortunately without success. However, during the process, I discovered something interesting:
If everything is placed directly on the View, the bug described above occurs. But if I insert a Text component inside the View, give the Text an accessible={true} prop, and leave the accessibility actions on the View, the problem disappears.
I'm not entirely sure why this happens. My best guess is that when using a Text component, the native build process adds different accessibility properties under the hood.
Here's my example code: [Snack](https://snack.expo.dev/@oaron/fix-for-accessibility-action-on-android)

**Comment by a-klotz-p8** [comment]
The only thing I was able to find out so far, is that the:
```
@Override
public boolean performAccessibilityAction(View host, int action, Bundle args) {
...
```
from the `ReactAccessibilityDelegate.java` is not even triggered when a accessibilityAction is triggered through double tap, but when triggered from the actions menu it is called just fine.

**Comment by oaron** [comment]
Thanks to the work of a developer I commissioned to investigate this issue in depth, a fix has been identified and submitted as a PR: https://github.com/facebook/react-native/pull/52724
I funded this effort independently in the hope that it will help move this long-standing accessibility bug forward for the benefit of the whole community. Looking forward to your review!

**Comment by meatnordrink** [comment]
One note for anyone else trying to address this before the fix makes it live (which won't happen until RN 0.86) - `patch-package` isn't a viable option here, as the React Native Android library is distributed as a precompiled AAR published to Maven Central - therefore `patch-package` has no effect. 

Just wanted to save others the trouble of trying it.

**Comment by oaron** [comment]
Just to add a data point: tested on a minimal Expo SDK 55 / RN 0.83.2 app with three items each exposing `edit` and `delete` custom accessibilityActions, built via gradle :app:assembleRelease (same AAR as EAS uses). TalkBack
swipe-up/down + double-tap fires `onAccessibilityAction` correctly for both
actions on all items — no "incompatible action". The fix from #52724 is already in the published `com.facebook.react:react-android:0.83.2` artifact (verified in the v0.83.2 tag), so it doesn't need to wait for 0.86.
